### PR TITLE
typescript: Improve error message for missmatched channels

### PR DESCRIPTION
**Pull request by defunctzombie**
Also fix the error message for message records missing a channel.

### CODE REVIEW

**Comment by defunctzombie** [comment]
I would actually recommend we remove the following checks from the reader _nextRecord_ call altogether:

```typescript
if (result.value?.type === "Channel") {
      const existing = this.channelsById.get(result.value.id);
      this.channelsById.set(result.value.id, result.value);
      if (existing && !isEqual(existing, result.value)) {
        throw new Error(
          `Channel record for id ${result.value.id} (topic: ${result.value.topic}) differs from previous for the same id.`,
        );
      }
    } else if (result.value?.type === "Message") {
      const channelId = result.value.channelId;
      const existing = this.channelsById.get(channelId);
      if (!existing) {
        throw new Error(`Encountered message on channel ${channelId} without prior channel record`);
      }
    }
```

Let the consumer handle this since they need to do the lookup anyway. They will be in a better position to ignore or error appropriately.

File: typescript/mcap/src/v0/Mcap0StreamReader.ts [hunk 1]
```diff
@@ -100,12 +100,15 @@ export default class Mcap0StreamReader implements McapStreamReader {
       const existing = this.channelsById.get(result.value.id);
       this.channelsById.set(result.value.id, result.value);
       if (existing && !isEqual(existing, result.value)) {
-        throw new Error(`differing channels for ${result.value.id}`);
+        throw new Error(
+          `Channel record for id ${result.value.id} (topic: ${result.value.topic}) differs from previous for the same id.`,
```
Comment (wkalt) [inline typescript/mcap/src/v0/Mcap0StreamReader.ts:104]: "from previous channel record..."?

Comment (defunctzombie) [inline typescript/mcap/src/v0/Mcap0StreamReader.ts:104]:
```suggestion
          `Channel record for id ${result.value.id} (topic: ${result.value.topic}) differs from previous channel record of the same id.`,
```

File: typescript/mcap/src/v0/Mcap0StreamReader.test.ts [hunk 2]
```diff
@@ -508,7 +508,9 @@ describe("Mcap0StreamReader", () => {
           messageEncoding: "utf12",
           metadata: new Map([["foo", "bar"]]),
         } as TypedMcapRecords["Channel"]);
-        expect(() => reader.nextRecord()).toThrow("differing channels for 42");
+        expect(() => reader.nextRecord()).toThrow(
+          /Channel record for id 42 \(topic: (myTopic|XXXXXXXX)\) differs from previous for the same id./,
```
Comment (defunctzombie) [inline typescript/mcap/src/v0/Mcap0StreamReader.test.ts:512]:
```suggestion
          /Channel record for id 42 \(topic: (myTopic|XXXXXXXX)\) differs from previous channel record of the same id./,
```

**Pull request merged**